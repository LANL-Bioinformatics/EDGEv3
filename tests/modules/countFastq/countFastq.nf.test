nextflow_workflow {

    name "Test Workflow COUNTFASTQ"
    script "modules/countFastq/countFastq.nf"
    workflow "COUNTFASTQ"

    test("1 Unpaired File") {

        when {
            params {
                shared {
                    pairedFile = false
                }
            }
            workflow {
                """
                input[0] = params.shared
                input[1] = file("${projectDir}/test_data/Ecoli_10x.1.fastq")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert snapshot(workflow.out).match()},
                {assert workflow.out.avgReadLen.get(0) == "151"}
            )
            
        }

    }

    test("1 Pair of PE files") {

        when {
            params {
                shared {
                    pairedFile = true
                }
            }
            workflow {
                """
                input[0] = params.shared
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert snapshot(workflow.out).match()},
                {assert workflow.out.avgReadLen.get(0) == "151"}
            )
            
        }

    }


}
