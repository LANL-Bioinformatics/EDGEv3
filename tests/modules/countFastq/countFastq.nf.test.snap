{
    "1 Pair of PE files": {
        "content": [
            {
                "0": [
                    "151"
                ],
                "1": [
                    [
                        "all.1.fastq:md5,c9cd5b2586c75c7448674ff3ff5451de",
                        "all.2.fastq:md5,3b69b59ed5c7111e5d57b900174ef2f6"
                    ]
                ],
                "avgReadLen": [
                    "151"
                ],
                "fastqFiles": [
                    [
                        "all.1.fastq:md5,c9cd5b2586c75c7448674ff3ff5451de",
                        "all.2.fastq:md5,3b69b59ed5c7111e5d57b900174ef2f6"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-06T13:53:45.128251613"
    },
    "1 Unpaired File": {
        "content": [
            {
                "0": [
                    "151"
                ],
                "1": [
                    "all.se.fastq:md5,c9cd5b2586c75c7448674ff3ff5451de"
                ],
                "avgReadLen": [
                    "151"
                ],
                "fastqFiles": [
                    "all.se.fastq:md5,c9cd5b2586c75c7448674ff3ff5451de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-06T13:51:42.65748549"
    }
}